x-pack/plugins/alerting/server/task_runner/task_runner_cancel.test.ts

```diff
@@ -0,0 +1,714 @@
+    id: '1',
+    consumer: 'bar',
+    createdAt: mockDate,
+    updatedAt: mockDate,
+    throttle: null,
+    muteAll: false,
+    notifyWhen: 'onActiveAlert',
+    enabled: true,
+    alertTypeId: ruleType.id,
+    apiKey: '',
+    apiKeyOwner: 'elastic',
+    schedule: { interval: '10s' },
+    name: 'rule-name',
+    tags: ['rule-', '-tags'],
+    createdBy: 'rule-creator',
+    updatedBy: 'rule-updater',
+    mutedInstanceIds: [],
+    params: {
+      bar: true,
+    },
+    actions: [
+      {
+        group: 'default',
+        id: '1',
+        actionTypeId: 'action',
+        params: {
+          foo: true,
+        },
+      },
+      {
+        group: RecoveredActionGroup.id,
+        id: '2',
+        actionTypeId: 'action',
+        params: {
+          isResolved: true,
+        },
+      },
+    ],
+    executionStatus: {
+      status: 'unknown',
+      lastExecutionDate: new Date('2020-08-20T19:23:38Z'),
+    },
+  };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    taskRunnerFactoryInitializerParams.getServices.mockReturnValue(services);
+    taskRunnerFactoryInitializerParams.getRulesClientWithRequest.mockReturnValue(rulesClient);
+    taskRunnerFactoryInitializerParams.actionsPlugin.getActionsClientWithRequest.mockResolvedValue(
+      actionsClient
+    );
+    taskRunnerFactoryInitializerParams.actionsPlugin.renderActionParameterTemplates.mockImplementation(
+      (actionTypeId, actionId, params) => params
+    );
+    ruleTypeRegistry.get.mockReturnValue(ruleType);
+    taskRunnerFactoryInitializerParams.executionContext.withContext.mockImplementation((ctx, fn) =>
+      fn()
+    );
+    rulesClient.get.mockResolvedValue(mockedRuleSavedObject);
+    encryptedSavedObjectsClient.getDecryptedAsInternalUser.mockResolvedValueOnce({
+      id: '1',
+      type: 'alert',
+      attributes: {
+        apiKey: Buffer.from('123:abc').toString('base64'),
+      },
+      references: [],
+    });
+    taskRunnerFactoryInitializerParams.actionsPlugin.isActionTypeEnabled.mockReturnValue(true);
+    taskRunnerFactoryInitializerParams.actionsPlugin.isActionExecutable.mockReturnValue(true);
+  });
+
+  test('updates rule saved object execution status and writes to event log entry when task is cancelled mid-execution', async () => {
+    const taskRunner = new TaskRunner(
+      ruleType,
+      mockedTaskInstance,
+      taskRunnerFactoryInitializerParams
+    );
+
+    const promise = taskRunner.run();
+    await Promise.resolve();
+    await taskRunner.cancel();
+    await promise;
+
+    const eventLogger = taskRunnerFactoryInitializerParams.eventLogger;
+    // execute-start event, timeout event and then an execute event because rule executors are not cancelling anything yet
+    expect(eventLogger.logEvent).toHaveBeenCalledTimes(3);
+    expect(eventLogger.startTiming).toHaveBeenCalledTimes(1);
+    expect(eventLogger.logEvent).toHaveBeenNthCalledWith(1, {
+      '@timestamp': '1970-01-01T00:00:00.000Z',
```

Comment:
There's a lot of duplication in these event structures - would it make sense to have a template we base these on, then customize based on the template?  Not clear to me if it would save much vertical space, or perhaps make the tests less readable.

Reply:
I actually started making a utility function to generate event log docs based on different parameters but I realized that I like being able to see the full shape of the expected event log doc when I am writing the tests (vs trying to figure it out based on what's being passed to the utility function), so I left it as is.

Reply:
Ya, I think I've had the same reaction trying to simplify things like this.  Kinda need a vscode tool to "fold excessive JSON" or something :-).